File: trunk/ZendAMF/app/servicos/ServicosUsuario.php
<?php
//include 'app/db/BaseDados.php';
include 'app/vo/UsuarioVO.php';

class ServicosUsuario {
	
	
	private $conn;
	
	function ServicosUsuario() {
			$db = new BaseDados();
			$this->conn = $db->conn;
	}
	
	public function addUsuario(UsuarioVO $usuario){
		
		$sql = "insert into usuario (nome,comissao,senha,permissao,login) values ('{$usuario->nome}',{$usuario->comissao},'{$usuario->senha}',{$usuario->permissao},'{$usuario->login}')";
		$resultado = $this->conn->Execute($sql);
		$usuario->codigo = $this->conn->insert_Id();
		return $usuario;
		
	}
	
	public function logar(UsuarioVO $usuario){
		$sql = "select * from usuario where login = '{$usuario->login}'";
		$resultado = $this->conn->Execute($sql);
		
		 if(!$resultado){
			 throw new Exception(Erros::$SQL_ERRO_MENSAGEM,Erros::$SQL_ERRO_CODIGO);
		 }
		 $dados_usuario = NULL;
		while($registro = $resultado->FetchNextObject()){			
			 $dados_usuario = $this->toUsuario($registro);
		}
		if($dados_usuario->senha==$usuario->senha){
			return $dados_usuario;
		}else{
			 throw new Exception(Erros::$AUTENTICACAO_FALHOU_MENSAGEM
			 ,Erros::$AUTENTICACAO_FALHOU_CODIGO); 
		}
		
	}
	
	
	public function removerUsuario(UsuarioVO $usuario){
		
		$sql = "delete from usuario  where codigo = {$usuario->codigo}";
		$resultado = $this->conn->Execute($sql);
		if($this->conn->Affected_Rows()==0){
			return false;
		}
		return true;
	}
	
	public function pesquisarUsuario($texto,$coluna){
		$sql = "select * from usuario where $coluna like '%$texto%'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$retorna_dados_usuario [] = $this->toUsuario($registro);
		}
		return $retorna_dados_usuario;
	}
	
	public function getUsuarios(){
		$sql = "select * from usuario";
		
		$resultado = $this->conn->Execute($sql);
		
		while($registro = $resultado->FetchNextObject()){			
			$retorna_dados_usuario [] = $this->toUsuario($registro);
		}
		return $retorna_dados_usuario;	
	}
	
	private function toUsuario($registro){
		if($registro->CODIGO != 0){
			$dados_usuario = new UsuarioVO();
			$dados_usuario->nome = $registro->NOME;
			$dados_usuario->codigo = $registro->CODIGO;
			$dados_usuario->comissao = (float)$registro->COMISSAO;
			$dados_usuario->permissao = $registro->PERMISSAO;
			$dados_usuario->senha = $registro->SENHA;
			$dados_usuario->login = $registro->LOGIN;
			return $dados_usuario;
		}
	}
}


//$eu = new ServicosUsuario();
//$usuario = new UsuarioVO();
//$usuario->login = "ADMIN";
//$usuario->senha = "admi";
//$servico = new ServicosUsuario();
//$eu->logar($usuario);
//$usuario->nome = "autoTesteTestado";
//$usuario->comissao = 10.98;
//$usuario->permissao = 1;
//$usuario->senha = "senha";
//$eu->addUsuario($usuario);
//$eu->getUsuarios();

?>

File: JavaServer/WEB-INF/src/services/ServicosUsuario.java
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Map;

import arquitetura.BancoDeDados;
import vo.UsuarioVO;

class ServicosUsuario {

	private BancoDeDados banco;

	public UsuarioVO addUsuario(UsuarioVO usuario) {

		String sql = "insert into usuario (nome,comissao,senha,permissao,login) values ('"
				+ usuario.nome
				+ "',"
				+ usuario.comissao
				+ ",'"
				+ usuario.senha
				+ "'," + usuario.permissao + ",'" + usuario.login + "')";

		if (banco.executarNoQuery(sql) == 0) {
			throw new RuntimeException("erro no addUsuario");
		}

		return usuario;

	}

	public UsuarioVO atualizarUsuario(UsuarioVO usuario) {
		String sql = "UPDATE usuario SET nome = '" + usuario.nome
				+ "' , comissao = " + usuario.comissao + " , senha = '"
				+ usuario.senha + "', permissao = " + usuario.permissao
				+ " , login  = '" + usuario.login
				+ "' where codigo = usuario.codigo";
		if (banco.executarNoQuery(sql) == 0) {
			throw new RuntimeException("erro no addUsuario");
		}
		return usuario;

	}

	public UsuarioVO logar(UsuarioVO usuario){
		String sql = "select * from usuario where login = 'usuario.login'";
		
		
		 if(!resultado){
			 throw new Exception(Erros::SQL_ERRO_MENSAGEM,Erros::SQL_ERRO_CODIGO);
	}
		 dados_usuario = NULL;
		 f = fopen('logou.txt','w+');
		 fwrite(f,dados_usuario.login);
		while(registro = resultado.FetchNextObject(){
			dados_usuario = new ();
			dados_usuario.nome = registro.NOME;
			dados_usuario.codigo = registro.CODIGO;
			dados_usuario.comissao = (float)registro.COMISSAO;
			dados_usuario.permissao = registro.PERMISSAO;
			dados_usuario.senha = registro.SENHA;
			dados_usuario.login = registro.LOGIN;
		}
		fwrite(f,dados_usuario.login);
		fwrite(f,dados_usuario.senha);
		if(dados_usuario.senha==usuario.senha){
			fwrite(f,'Entrou');
			return dados_usuario;
		}else{
			 throw new Exception(Erros::AUTENTICACAO_FALHOU_MENSAGEM
			 ,Erros::AUTENTICACAO_FALHOU_CODIGO); 
		}
		
		}

	public boolean removerUsuario(UsuarioVO usuario) {

		String sql = "delete from usuario  where codigo = " + usuario.codigo
				+ "";

		if (this.banco.executarNoQuery(sql) == 0) {
			return false;
		}
		return true;
	}

	public Map<Integer, UsuarioVO> pesquisarUsuario(String texto, String coluna) {
		String sql = "select * from usuario where " + coluna + " like '%"
				+ texto + "%'";
		ResultSet rs = banco.executar(sql);
		Map<Integer, UsuarioVO> usuarios = null;
		int i = 0;
		try {
			while (rs.next()) {

				UsuarioVO dados_item = new UsuarioVO();
				dados_item.codigo = rs.getInt("codigo");
				dados_item.nome = rs.getString("nome");
				dados_item.comissao = rs.getDouble("comissao");
				dados_item.login = rs.getString("login");
				dados_item.permissao = rs.getInt("permissao");
				usuarios.put(i, dados_item);
				i++;
			}
		} catch (SQLException e) {
			System.out.println("erro");
		}
		return usuarios;
	}

	public Map<Integer, UsuarioVO> getUsuarios() {
		String sql = "select * from usuario";

		ResultSet rs = banco.executar(sql);
		Map<Integer, UsuarioVO> usuarios = null;
		int i = 0;
		try {
			while (rs.next()) {

				UsuarioVO dados_item = new UsuarioVO();
				dados_item.codigo = rs.getInt("codigo");
				dados_item.nome = rs.getString("nome");
				dados_item.comissao = rs.getDouble("comissao");
				dados_item.login = rs.getString("login");
				dados_item.permissao = rs.getInt("permissao");
				usuarios.put(i, dados_item);
				i++;
			}
		} catch (SQLException e) {
			System.out.println("erro");
		}
		return usuarios;
	}

}
